index.bs

```diff
@@ -488,13 +490,238 @@ partial interface NeuralNetworkContext {
     </div>
 </div>
 
+### gru ### {#api-neuralnetworkcontext-gru}
+Gated Recurrent Unit [[GRU]] recurrent network using an update gate and a reset gate to compute the hidden state that rolls into the output across the temporal sequence of the Network
+<script type=idl>
+enum RecurrentNetworkDirection {
+  "forward",
+  "backward",
+  "both"
+};
+
+enum RecurrentNetworkWeightLayout {
+  "zrn",  // update-reset-new gate ordering
+  "rzn"   // reset-update-new gate ordering
+};
+
+enum RecurrentNetworkActivation {
+  "relu",
+  "sigmoid",
+  "tanh"
+};
+
+partial interface NeuralNetworkContext {
+  sequence<Operand> gru(Operand input, Operand weight, 
+                        Operand recurrentWeight, Operand initialHiddenState,
+                        long hiddenSize, long steps, 
+                        optional Operand bias, optional Operand recurrentBias,
+                        optional boolean resetAfter = true,
+                        optional boolean returnSequence = false,
+                        optional RecurrentNetworkDirection direction = "forward",
+                        optional RecurrentNetworkWeightLayout layout = "zrn",
+                        optional sequence<RecurrentNetworkActivation> activations);
+};
+</script>
+<div algorithm=gru>
+    **Arguments:**
+        - *input*: an {{Operand}}. The input 3-D tensor of shape [steps, batch_size, input_size]. 
+        - *weight*: an {{Operand}}. The 3-D input weight tensor of shape [num_directions, 3 * hidden_size, input_size]. The ordering of the weight vectors in the second dimension of the tensor shape is specified according to the *layout* argument.
+        - *recurrentWeight*: an {{Operand}}. The 3-D recurrent weight tensor of shape [num_directions, 3 * hidden_size, hidden_size]. The ordering of the weight vectors in the second dimension of the tensor shape is specified according to the *layout* argument.
+        - *initialHiddenState*: an {{Operand}}. The 3-D initial hidden state tensor of shape [num_directions, batch_size, hidden_size].
+        - *hiddenSize*: a {{long}} scalar. The value of the third dimension of the cell output tensor shape. It indicates the number of features in the hidden state.
+        - *steps*: a {{long}} scalar. The number of time steps in the recurrent network.
```

Comment:
What about varying length sequences? Do we want to support them?

Reply:
I can't find other use of that in PyTorch or TensorFlow. It seems something specific to ONNX. Do you have more context of why it is needed besides it is defined in ONNX?

Reply:
@gramalingam I did some research and I couldn't find variable length sequence supported in any other popular frameworks but ONNX. I looked at TensorFlow (and its JS sibling), PyTorch, CoreML and MXNet. No other framework supports this but ONNX. Do you remember why ONNX added this? Can we leave it for now for compactness?

Reply:
It possibly comes from CNTK.